cli/azd/cmd/env.go

```diff
@@ -592,6 +598,104 @@ func (eg *envGetValuesAction) Run(ctx context.Context) (*actions.ActionResult, e
 	return nil, eg.formatter.Format(env.Dotenv(), eg.writer, nil)
 }
 
+func newEnvGetValueFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *envGetValueFlags {
+	flags := &envGetValueFlags{}
+	flags.Bind(cmd.Flags(), global)
+
+	return flags
+}
+
+func newEnvGetValueCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-value <keyName>",
+		Short: "Get specific environment value.",
+	}
+	cmd.Args = cobra.MaximumNArgs(1)
+
+	return cmd
+}
+
+type envGetValueFlags struct {
+	internal.EnvFlag
+	global *internal.GlobalCommandOptions
+}
+
+func (eg *envGetValueFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	eg.EnvFlag.Bind(local, global)
+	eg.global = global
+}
+
+type envGetValueAction struct {
+	azdCtx     *azdcontext.AzdContext
+	console    input.Console
+	envManager environment.Manager
+	writer     io.Writer
+	flags      *envGetValueFlags
+	args       []string
+}
+
+func newEnvGetValueAction(
+	azdCtx *azdcontext.AzdContext,
+	envManager environment.Manager,
+	console input.Console,
+	writer io.Writer,
+	flags *envGetValueFlags,
+	args []string,
+
+) actions.Action {
+	return &envGetValueAction{
+		azdCtx:     azdCtx,
+		console:    console,
+		envManager: envManager,
+		writer:     writer,
+		flags:      flags,
+		args:       args,
+	}
+}
+
+func (eg *envGetValueAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	if len(eg.args) < 1 {
+		return nil, fmt.Errorf("no key name provided")
+	}
+
+	keyName := eg.args[0]
+
+	name, err := eg.azdCtx.GetDefaultEnvironmentName()
```

Comment:
You likely copied/pasted this from another command but we really shouldn't need to do all this extra work around default environments, yada, yada.  Instead you should be able to just inject the `*environment.Environment` into the action and pull the value out.

Reply:
injecting `*environment.Environment` comes with a high price!  It will be added to the current tech-debt of the Lazy[T] to keep it working.

I would recommend keeping it like this as a cleaner and easier way to understand the flow of the environment creation by using the envManager.

Reply:
@wbreza and @vhvb1989 what did you decide?

Reply:
I'd probably wait for @weikanglim  and @ellismg  here.

Reply:
Inject environment wouldn't work. Here we want specifically to "error out" instead of "initialize environment"

`azd env get-values ladeedaathisenvdoesnotexist`

Expect:
"environment 'ladeedaathisenvdoesnotexist' does not exist'

Not:
"Would you like to create environment 'ladeedaathisenvdoesnotexist' ?"

We went this same route previously with https://github.com/Azure/azure-dev/commit/4d00b47e9e6314e627f7e760182afa653b26d684